src/basic/cgroup-util.c

```diff
@@ -2052,8 +2052,13 @@ int cg_unified_cached(bool flush) {
                         unified_cache = CGROUP_UNIFIED_SYSTEMD;
                         unified_systemd_v232 = false;
                 } else {
-                        if (statfs("/sys/fs/cgroup/systemd/", &fs) < 0)
-                                return log_debug_errno(errno, "statfs(\"/sys/fs/cgroup/systemd\" failed: %m");
+                        if (statfs("/sys/fs/cgroup/systemd/", &fs) < 0) {
+                                log_debug_errno(errno, "statfs(\"/sys/fs/cgroup/systemd\" failed: %m");
+                                if (errno == ENOENT)
+                                        /* Some other software may have set up /sys/fs/cgroup in a configuration we do not recognize. */
+                                        return log_debug_errno(SYNTHETIC_ERRNO(ENOMEDIUM), "No known cgroup or cgroup2 hierarchy found.");
```

Comment:
hmm, i think we should only log about the error once. But in case you now log twice. change to:

```c
if (statfs(…) < 0) {
        if (errno == ENOENT) {
                log_debug_errno(errno, "Unsupported cgroupsv1 setup detected: name=systemd hierarchy not found.");
                return -ENOMEDIUM;
        }
        return log_debug_errno(errno, "statfs(\"/sys/fs/cgroup/systemd\" failed: %m");
}
```